activity_browser/ui/tables/plugins.py

```diff
@@ -36,23 +37,24 @@ def mousePressEvent(self, e):
             proxy = self.indexAt(e.pos())
             if proxy.column() == 0:
                 new_value = not bool(proxy.data())  
-                plugin_name = self.model.get_plugin_name(proxy)
-                if new_value:
-                    signals.plugin_selected.emit(plugin_name)
-                else:
+                plugin_name = self.model.get_plugin_name(proxy) #
+                if not new_value:
                     msgBox = QMessageBox()
                     msgBox.setText("Remove plugin from project ?")
                     msgBox.setInformativeText("This will removed all data created by the plugin.")
                     msgBox.setStandardButtons(QMessageBox.Ok | QMessageBox.Cancel)
                     msgBox.setDefaultButton(QMessageBox.Cancel)
                     ret = msgBox.exec_()
-                    if ret == QMessageBox.Ok:
-                        signals.plugin_deselected.emit(plugin_name)
-                self.model.sync()
-        super().mousePressEvent(e)
+                    if ret == QMessageBox.Cancel:
+                        new_value = not new_value
+                self.model.sync(proxy, new_value) # TODO check this usage it is currently employed to ensure synchronization
```

Comment:
Should these 2 TODO be finished before merge?

Reply:
I've just checked both TODO messages, both are good for committing